kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -164,7 +183,25 @@ internal class KotlinJsonAdapter<T>(
   }
 }
 
-class KotlinJsonAdapterFactory : JsonAdapter.Factory {
+/**
+ * Failure strategy when a required value is missing or non-null value is null
+ */
+enum class FailureStrategy {
```

Comment:
We may want to extract this to the main moshi artifact for shared use in code gen